build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/datastore/DatastoreAdapter.java

```diff
@@ -72,7 +73,9 @@ public BuildConfigSetRecord saveBuildConfigSetRecord(BuildConfigSetRecord buildC
      * @return The latest revision of the given build configuration
      */
     public BuildConfigurationAudited getLatestBuildConfigurationAudited(Integer buildConfigurationId) {
-        return datastore.getLatestBuildConfigurationAudited(buildConfigurationId);
+        BuildConfigurationAudited buildConfigAudited = datastore.getLatestBuildConfigurationAudited(buildConfigurationId);
+        buildConfigAudited.getProject().getBuildConfigurations().forEach(BuildConfiguration::getId);
```

Comment:
I've just tested it. 

`getBuildConfigurations()` is not enough, `LazyInitializationException` is still thrown on accessing the `buildConfigurations` of `Project`.

Long story: AFAIR, we put the stuff that is fetched here in `submittedBuilds`. Then, later, handling a different request, we take a build from it and try to get build configurations from the build.